### DISCUSSION TITLE
Make `camelCase` names more readable

### DISCUSSION THREAD

**Discussion by babakfp** [tonsky/FiraCode#1573] [Ideas]
Hi

Can we hame some sort of a design that makes `camelCase` names more readable?

Maybe make the uppercase letters bigger, or add a bit more space from the left.

Can we write in `camelCase` but read in `snake_case`? that sort of a feature would be awesome!

Does it sound interesting? Can we please make this happen?

Thanks

**Comment by cormullion** [comment]
> add a bit more space from the left

Technically OpenType has tools to do this (eg here's the results of adding the rule `pos @lowercase @uppercase 250;`:


before

<img width="384" alt="Screenshot 2024-02-14 at 08 56 20" src="https://github.com/tonsky/FiraCode/assets/52289/dcc352ea-fd45-436e-a7ad-9e4349cc3c71">

after

<img width="445" alt="Screenshot 2024-02-14 at 08 55 31" src="https://github.com/tonsky/FiraCode/assets/52289/8a40404d-43d1-4d98-9885-8ed3402798d0">

But it would work in some environments (eg VS Code) but not in others... And it would introduce irregularities- eg columns in tables would no longer align. 

> Maybe make the uppercase letters bigger

This would require a complete set of slightly larger letters to be added to the font, I think?

> Can we write in camelCase but read in snake_case?

Sounds like it could be very confusing...

Perhaps instead you could try increasing the character spacing on your terminal.

**Reply by babakfp** [reply]
> before
> 
> <img alt="Screenshot 2024-02-14 at 08 56 20" width="384" src="https://private-user-images.githubusercontent.com/52289/304676907-dcc352ea-fd45-436e-a7ad-9e4349cc3c71.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NjQ3NDIsIm5iZiI6MTcwNzk2NDQ0MiwicGF0aCI6Ii81MjI4OS8zMDQ2NzY5MDctZGNjMzUyZWEtZmQ0NS00MzZlLWE3YWQtOWU0MzQ5Y2MzYzcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDAyMzQwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY3MDkxY2IzMTBmZjcxZTQ2ZjczMDBhODRjZWNmYTIwNWVjYTU3MWNkYTNhNTdjNzdlN2U0YzJmNjIwNjgwN2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.byxgCZmeU1ccMHMxhyooI-Ft5WMn0ES7AC9gMQ4Vy0E">
> after
> 
> <img alt="Screenshot 2024-02-14 at 08 55 31" width="445" src="https://private-user-images.githubusercontent.com/52289/304676919-8a40404d-43d1-4d98-9885-8ed3402798d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NjQ3NDIsIm5iZiI6MTcwNzk2NDQ0MiwicGF0aCI6Ii81MjI4OS8zMDQ2NzY5MTktOGE0MDQwNGQtNDNkMS00ZDk4LTk4ODUtOGVkMzQwMjc5OGQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDAyMzQwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjMTBmY2I1NmJmMjRkZTRhY2NhZTIwZTY4OTFlNTRkNmY2MTBjMDNkOTNkNjIzNzU1ZWQ3Yzc0NmM3Njc3YTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IZhfxMUObwSixRX8drhsyeCwdalu0ZMYpLR2tPM_axw">

Hmm, I couldn't even tell the difference. This proves to me that it wasn't a good idea. Even if the spacing was more, then it would have looked like separate words. Thanks.

Take a look at this https://github.com/tonsky/FiraCode/discussions/1573#discussioncomment-8474490, I think it's beautiful, and this way we won't be having the issues of the other ideas.

**Comment by tonsky** [comment]
What you are looking for is a font with low x-height. It’s really hard to retrofit into existing font, so it’s better to find one that fits you needs already

**Reply by babakfp** [reply]
I searched for that and found this:

![image](https://github.com/tonsky/FiraCode/assets/44144724/e24f1802-e619-4c9f-9cc1-d82d01fee14d)

Something like this ^^^ would be fantastic. Look at how big the `O` and `S` are, that is so readable and I love it.

Would it be possible to add it to FiraCode?

Are there any mono-spaced fonts that are as exaggerated as the above font?

**Comment by cormullion** [comment]
>Are there any mono-spaced fonts that are as exaggerated as the above font?

Unlikely... ! The best you can probably do is [Latin Modern Mono](https://www.programmingfonts.org/#latin-modern), which is  based on an old-school font (and is pretty ugly too :)). Modern monospaced fonts tend to increase x-height for readability. The font with the best contrast would probably be Iosevka.

**Reply by babakfp** [reply]
Yeah, it's ugly 😄.

> Modern monospaced fonts tend to increase x-height for readability.

In a design file, I tried to increase the height of uppercase letters and I realized that that my eyes needed to go up to be able to read! Maybe I designed it badly, but I'm not sure.

I'll try "Iosevka" out. Thanks.

**Reply by alexeyten** [reply]
@babakfp 
Iosevka could do this :smile: 
![image](https://github.com/tonsky/FiraCode/assets/44377/2204d50c-4fec-4c66-878b-912848cee120)
